src/Raven.Studio/typescript/viewmodels/database/query/spatialQueryMap.ts

```diff
@@ -0,0 +1,104 @@
+
+        const otmLink = `<a href="http://opentopomap.org/">OpenTopoMap</a>`;
+        const otmUrl = `http://{s}.tile.opentopomap.org/{z}/{x}/{y}.png`;
+        const otmAttrib = `&copy; ${otmLink} Contributors`;
+        const otmMap = L.tileLayer(otmUrl, {attribution: otmAttrib});
+
+        const baseLayers = {
+            "Streets Map": osmMap,
+            "Topography Map": otmMap
+        };
+
+        //////////////////////////////////////////////////////////////
+        // Define group layers (layer per spatial point from query)
+        //////////////////////////////////////////////////////////////
+
+        let dataLayers = {};
+        let markersGroups: MarkerClusterGroup[] = [];
+
+        this.geoData().forEach((layerModel) => {
+            
+            let markers: Layer[] = [];
+            let markersGroup = L.markerClusterGroup();
+
+            const generatePopupContent = (doc: document) => {
+                const docDto = doc.toDto(true);
+                const metaDto = docDto["@metadata"];
+                documentMetadata.filterMetadata(metaDto);
+
+                let text = JSON.stringify(docDto, null, 4);
+                text = Prism.highlight(text, (Prism.languages as any)["javascript"]);
+
+                const textHtml = `<div>
+                                    <h4>Document: ${doc.getId()}</h4>
+                                    <hr>
+                                    <pre>${text}</pre>
+                                 </div>`;
+                return textHtml;
+            }
+
+            layerModel.geoPoints().forEach((point) => {
+                const pointMarker = L.marker([point.latitude, point.longitude], { title: point.popupContent.getId() })
+                                     .bindPopup(generatePopupContent(point.popupContent), { 'className' : 'custom-popup', 'maxWidth': 600, 'maxHeight': 400 } );
+                markers.push(pointMarker);
+            });
+
+            markersGroup.addLayers(markers);
+            const controlLayerText = `<span>Point fields:</span>
+                                      <div class='margin-left'>
+                                          <span>${layerModel.latitudeProperty}</span><br>
+                                          <span>${layerModel.longitudeProperty}</span>
+                                      </div>`;
+            (dataLayers as any)[controlLayerText] = markersGroup;
+            markersGroups.push(markersGroup);
+        })
+
+        //////////////////////
+        // Define the map 
+        //////////////////////
+
+        let map = L.map('mapid', {preferCanvas: true});      // init map
+        osmMap.addTo(map);                                   // init base layer to use
+        L.control.layers(baseLayers, dataLayers).addTo(map); // init legend control
+        markersGroups.forEach(group => map.addLayer(group)); // init markers groups
+        
+        let bounds = markersGroups[0].getBounds();
```

Comment:
const?

Reply:
can you start with empty bound L.bounds([])? 
Then we don;t need special treatment for first and other elements?

Reply:
`map.fitBounds()` expects a `LatLngBoundsExpression `expression which is **either** what you get from a `markersGroup.getBounds()` method and then you can extend it to include bounds of another group,
**or** a plain list of all the points in the data.

So it is **either** doing it as I did here where each group bounds extends the other, 
**or** making something like:

```
let allPoints: any[] = [];
        this.geoData().forEach(x => {
            x.geoPoints().forEach(y => {
                allPoints.push([y.latitude, y.longitude]);
            })
        })

map.fitBounds(allPoints, {padding: [50, 50]});
```
`L.bounds()` returns `Bounds` which represents a rectangular area in pixel coordinates
and is Not passed to` map.fitBounds`

Reply:
?

Reply:
ok - done - with `L.latLngBounds([];`